src/channel/fs_prover_channel.rs

```diff
@@ -0,0 +1,112 @@
+use crate::channel::{Channel, ChannelStates, ProverChannel}; // ProverChannel 추가
+use crate::hashutil::TempHashContainer;
+use ark_ff::Field;
+use std::marker::PhantomData;
+
+use rand_chacha::rand_core::RngCore;
+use rand_chacha::ChaCha20Rng;
+
+pub struct FSProverChannel<F: Field, H: TempHashContainer> {
+    _ph: PhantomData<(F, H)>,
+    prng: ChaCha20Rng,
+    proof: Vec<u8>,
+    states: ChannelStates,
+}
+
+impl<F: Field, H: TempHashContainer> AsMut<ChannelStates> for FSProverChannel<F, H> {
```

Comment:
This implementation was created because I wanted to enforce that the `ChannelState` must be included in the struct when implementing the `Channel` trait.

It seems that the problem becomes simpler if I keep the `Channel` trait and `ChannelStatus` trait separate.
No needs to expose `ChannelStates` at mut then.